.github/workflows/rust.yml

```diff
@@ -51,6 +53,28 @@ jobs:
     - run: rustup update
     - run: rustup target add ${{ matrix.sys.target }}
     - run: cargo install --locked --version 0.5.14 cargo-hack
-    - run: cargo hack --feature-powerset check --target ${{ matrix.sys.target }}
+    - run: cargo hack --feature-powerset check --locked --target ${{ matrix.sys.target }}
     - if: matrix.sys.test
-      run: cargo hack --feature-powerset test --target ${{ matrix.sys.target }}
+      run: cargo hack --feature-powerset test --locked --target ${{ matrix.sys.target }}
+
+  cargo-publish:
+    if: github.event_name == 'release'
+    needs: [complete]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - run: rustup update
+    - run: cargo publish --locked
+      env:
+      - CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+
+  cargo-publish-dry-run:
+    if: github.event_name == 'push' && startsWith(github.ref_name, 'release-dry-run')
+    needs: [complete]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - run: rustup update
+    - run: cargo publish --locked --dry-run
+      env:
+      - CARGO_REGISTRY_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
```

Comment:
Maybe we should dry run with multiple targets, like x86 linux and wasm32, as well as run the publish dry run with cargo hack. I think the only check that the versions in the dependencies (and not the git URLs) are correct, will be done by the publish, which means we should probably check all the features too.